### PR TITLE
Modules

### PR DESCRIPTION
**Pull request by Kha**
As expected, a bit more work than expected.

* Compile modules to CommonJS via babel, then use webpack to merge them
  and inline the view templates => drop infuser, koExternalTemplateEngine

* Slim down awful App class, use dynamically selected components for navigation

* Remove that weird 'cart' navigation item

### CODE REVIEW

**Comment by Kha** [comment]
rebased

File: views/login.html [hunk 1]
```diff
@@ -1,21 +1,21 @@
 <!-- login screen template
-Data binding context: App -->
+Data binding context: User -->
 
 <div class="row">
-  <h1 class="banner">Einmal anmelden. Alle fsmi Produkte nutzen.</h1>
+  <h1 class="banner">Einmal anmelden. Alle fsmi-Produkte nutzen.</h1>
```
Comment (TehMillhouse) [inline views/login.html]: B-but Google has a Deppenleerzeichen on the login page, too. :(

**Comment by TehMillhouse** [comment]
The components thing and the changes made to the templates are nice. I'm glad they've finally got the right context.  
I'm a bit confused by many small things (ES6 stuff, the loaders for `main.js`, what exactly the `default` keyword entails for exports, or why our `post` utility function is in `config`), but assuming these things are there for a good reason (or, more realistically, a really bone-headed but unavoidable reason),  
LGTM.

File: views/login.html (hunk 1 above)
Comment (Kha) [inline views/login.html]: ...couldn't resist :(

**Comment by Kha** [comment]
No comment on the `main.js` loader :D . `post` is in `config` simply because `baseUrl` is there. I thought about other (most likely better) designs such as putting them in a ViewModel base class, but it really should be part of the model (which still doesn't exist).

**Pull request merged**